src/utils/getFolderWithReference.js

```diff
@@ -0,0 +1,66 @@
+import { APPS_DOCTYPE, FILES_DOCTYPE } from 'src/doctypes'
+import { getClient } from 'src/utils/client'
+
+const APP_DIR_NAME = 'Papers'
+const ADMINISTRATIVE_DIR_REF = `${APPS_DOCTYPE}/administrative`
+const APP_DIR_REF = `${APPS_DOCTYPE}/papers`
+
+const getAdministrativeFolderWithReference = async client => {
+  const fileCollection = client.collection(FILES_DOCTYPE)
+  const { included } = await fileCollection.findReferencedBy({
+    _id: ADMINISTRATIVE_DIR_REF,
+    _type: APPS_DOCTYPE
+  })
+
+  return included
+}
+
+const getAppFolderWithReference = async client => {
+  const fileCollection = client.collection(FILES_DOCTYPE)
+  const { included } = await fileCollection.findReferencedBy({
+    _id: APP_DIR_REF,
+    _type: APPS_DOCTYPE
+  })
+
+  return (
+    (included &&
+      included.filter(
+        folder => /^\/\.cozy_trash/.test(folder.path) === false
+      )) ||
+    []
+  )
+}
+
+const getOrCreateAppFolderWithReference = async () => {
+  const client = getClient()
+  const existingFolders = await getAppFolderWithReference(client)
+
+  if (existingFolders.length > 0) {
+    return existingFolders[0]
+  } else {
+    const fileCollection = client.collection(FILES_DOCTYPE)
+    const [
+      { name: administrativeName }
+    ] = await getAdministrativeFolderWithReference(client)
+
+    const dirId = await fileCollection.ensureDirectoryExists(
+      `/${administrativeName}/${APP_DIR_NAME}`
+    )
+    await fileCollection.addReferencesTo(
+      {
+        _id: APP_DIR_REF,
+        _type: APPS_DOCTYPE
+      },
+      [
+        {
+          _id: dirId
+        }
+      ]
+    )
+    const { data: dirInfos } = await fileCollection.get(dirId)
+
+    return dirInfos
+  }
+}
+
+export default getOrCreateAppFolderWithReference
```

Comment:
You should use (and update) https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/models/folder.js